jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
+    /// This field may include role mappings that help define the access levels for issued tokens.
+    #[serde(deserialize_with = "check_token_metadata")]
+    pub token_metadata: Option<Vec<TokenMetadata>>,
+}
 
-    use std::collections::HashMap;
+/// Validates the `token_metadata` field.
+///
+/// This function ensures that the metadata contains at most one `TokenMetadata` with a `role_mapping`
+/// to prevent ambiguous role assignments.
+fn check_token_metadata<'de, D>(deserializer: D) -> Result<Option<Vec<TokenMetadata>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let token_metadata: Option<Vec<TokenMetadata>> = Option::deserialize(deserializer)?;
 
-    use base64::prelude::*;
-    use cedar_policy::PolicyId;
+    if let Some(metadata) = &token_metadata {
+        let count = metadata
+            .iter()
+            .filter(|token| token.role_mapping.is_some())
+            .count();
 
-    // we use camel case to show that it is like a constant
-    #[derive(Debug, thiserror::Error)]
-    enum ParsePolicySetMessage {
-        #[error("unable to decode policy_content as base64")]
-        Base64,
-        #[error("unable to decode policy_content to utf8 string")]
-        String,
-        #[error("unable to decode policy_content from human readable format")]
-        HumanReadable,
-        #[error("could not collect policy store's to policy set")]
-        CreatePolicySet,
+        if count > 1 {
+            return Err(serde::de::Error::custom(
+                "there can only be one TokenMetadata with a role_mapping".to_string(),
+            ));
+        }
     }
 
-    /// Represents a raw data of the `Policy` in the `PolicyStore`
-    /// is private and used only in the [`parse_policy_set`] function
-    #[derive(Debug, serde::Deserialize)]
-    struct RawPolicy {
-        pub policy_content: String,
+    Ok(token_metadata)
+}
+
+/// Represents metadata associated with a token.
+///
+/// This struct includes the type of token, the ID of the person associated with the token,
+/// and an optional role mapping for access control.
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
+pub struct TokenMetadata {
+    /// The type of token (e.g., Access, ID, Userinfo, Transaction).
+    #[serde(rename = "type")]
+    pub kind: TokenKind,
+
+    /// The claim used to create the person entity associated with this token.
+    pub person_id: String,
+
+    /// An optional claim used to create a role for the token.
+    pub role_mapping: Option<String>,
+}
+
+#[derive(Debug, Copy, Clone)]
+#[allow(dead_code)]
+pub enum TokenKind {
+    /// Access token used for granting access to resources.
+    Access,
+
+    /// ID token used for authentication.
+    Id,
+
+    /// Userinfo token containing user-specific information.
+    Userinfo,
+
+    /// Transaction token used for tracking transactions.
+    Transaction,
+}
+
+/// Enum representing the different kinds of tokens used by Cedarling.
+impl fmt::Display for TokenKind {
```

Comment:
Just for case, we can use [`derive_more::Display`](https://crates.io/crates/derive_more). But current implementation also OK.